### DISCUSSION TITLE
Can't support tor SOCKS?

### DISCUSSION THREAD

**Discussion by tripbnb66** [spatie/browsershot#482] [Q&A] [answered]
Hi

I enabled tor (https://www.torproject.org/) on my Mac OSx and execute my simple php script but it failed and got a error message "Socks version 67 not recognized. (This port is not an HTTP proxy; did you want to use HTTPTunnelPort?)".
Is there a API of browsershot that can support SOCKS?

$ret = Browsershot::url($url)
    ->setIncludePath('$PATH:/usr/local/bin')
    ->setNodeBinary('/usr/local/bin/node')
    ->setNpmBinary('/usr/local/bin/npm')
    ->setProxyServer('http://127.0.01:9050')   # this line doesn't work for tor, is there a setSocksServer or similar API? 
    ->setExtraHttpHeaders(['User-Agent' => $user_agent])
    ->bodyHtml();
var_dump($ret);

**Comment by pgrimaud** [comment, marked as answer]
If you're using SOCKS 5 proxy, you should try : 

```php
->setProxyServer('socks5://127.0.01:9050')
```

According to 
- https://www.chromium.org/developers/design-documents/network-settings#TOC-Command-line-options-for-proxy-settings
- https://www.chromium.org/developers/design-documents/network-stack/socks-proxy

**Comment by tripbnb66** [comment]
@pgrimaud  thanks to your help and your solutions works.
